enhancements/pipeline-tooling.md

```diff
@@ -0,0 +1,267 @@
+# Pipeline Authors have Tools to build, modify, and consume the latest bundle and registry formats
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift/docs]
+
+## Summary
+
+There are several internal pipelines that deal with OLM operator data, all of which need to build or consume the newer bundle and registry formats. We can build new bundle and registry formats as per the other 4.4 epics, but if no one internally is using them we will not derive much of the desired value from them.
+
+### Goals
+
+* Maintainers of Operator Registries can leverage container registries so that they can host Operator bundles and indexes of all available bundles in this catalog in the form of container images
+* Maintainers of Operator Registries can use a tool to:
+    * create a bundle index so that it is easy to employ in a CI environment.
+    * migrate to and from both old and new format.
+    * merge indexes
+    * delete the latest version of an operator from an index
+    * download the contents of index images
+    * specify related images for CVE updates.
+    * validate the contents of an index before publishing it.
+    * package/release/version tool(s)
+* Pipeline admins can publish multiple applications simultaneously without conflicts.
+* Developers or QE can use downstream images in a dev/stage/qe repository
+* There is sufficient documentation for operator owners and maintainers of operatory registries.
+* Design the migration of Community Operators to the new bundle format.
+
+### Non-Goals
+
+* Migrate Community Operators / operatorhub.io to the new bundle format.
+* Pipeline admins rely on the tooling to extrapolate defined selective metadata from full set of each application operator manifests and include that in a combined catalog for index object
+* Pipeline admins can update official catalog object with newly created single application operator and publish
+
+---
+
+## User Stories
+
+### Migrate to and from each registry format
+
+As an Operator Registry maintainer, I want to continue to push updates to old (appregistry-based) catalogs after having migrated to the new bundle format without having to maintain two repositories.
+
+#### Design Proposal
+
+Add a command to `opm index` and `opm alpha bundle` called `migrate` which:
+
+* takes a collection of bundles migrates these to the old manifest format.
+
+    **Example CLI:**
+    `opm alpha bundle migrate --bundles quay.io/my/bundle:v1,quay.io/my/bundle:v2 --out path/to/my/dir`
```

Comment:
How does this work if the channel data is ordered? We will need to construct the channel update graphs in order to synthesize the `package.yaml` file for the old manifest format. Can we go into some detail on how that works in this proposal?